### PR TITLE
update RELEASE_NOTES for 0.8.0

### PR DESCRIPTION
**Pull request by qiaojialin**
update release_notes

### CODE REVIEW

File: RELEASE_NOTES [hunk 1]
```diff
@@ -4,36 +4,148 @@ Apache IoTDB (incubating) 0.8.0
...
+Enable users define the location of their thrift compiler
+Add example module
+Add a log appender: put info, warn, error log into one file and disable log_info by default
+Recover when resource file does not exist while tsfile is complete
+
 
 Incompatible changes
 --------------------
 
+Data folder structures
```
Comment (jixuan1989) [inline RELEASE_NOTES]: You'd better to claim clearly be incompatible with WHO?

Comment (JulianFeinauer) [reply RELEASE_NOTES]: Wat is WHO?

**Comment by JulianFeinauer** [comment]
I cannot comment on all things in detail but the update looks excellent and extensive... but I think @jixuan1989 should merge it, when his comment is adressed.

File: RELEASE_NOTES (hunk 1 above)
Comment (qiaojialin) [reply RELEASE_NOTES]: He means with which version... I have updated this description

Comment (jixuan1989) [reply RELEASE_NOTES]: +1.

**Pull request merged**